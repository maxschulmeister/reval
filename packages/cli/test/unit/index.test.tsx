import { describe, it, expect } from 'vitest';
import React from 'react';
import { render } from 'ink-testing-library';
import Index from '../../src/commands/index';

describe('Index (Help) Command', () => {
  it('renders main help screen with commands list', () => {
    const { lastFrame } = render(<Index options={{}} />);
    const output = lastFrame();
    
    expect(output).toContain('ðŸŽ¯ reval CLI v0.1.0');
    expect(output).toContain('A benchmark framework for evaluating LLM applications');
    expect(output).toContain('Commands:');
    expect(output).toContain('run');
    expect(output).toContain('list');
    expect(output).toContain('show');
    expect(output).toContain('export');
    expect(output).toContain('init');
    expect(output).toContain('ui');
    expect(output).toContain('version');
    expect(output).toContain('Database Commands:');
    expect(output).toContain('db create');
    expect(output).toContain('db migrate');
    expect(output).toContain('db studio');
    expect(output).toContain('Quick Start:');
  });

  it('renders version when --version flag is provided', () => {
    const { lastFrame } = render(<Index options={{ version: true }} />);
    
    expect(lastFrame()).toBe('reval CLI v0.1.0');
  });

  it('help screen matches snapshot', () => {
    const { lastFrame } = render(<Index options={{}} />);
    
    expect(lastFrame()).toMatchSnapshot();
  });
});