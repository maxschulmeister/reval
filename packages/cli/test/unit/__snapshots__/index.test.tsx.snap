// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Index (Help) Command > help screen matches snapshot 1`] = `
"ðŸŽ¯ reval CLI v0.1.0
A benchmark framework for evaluating LLM applications
Commands:
  run     Execute a benchmark run
  list    List recent benchmark runs
  show    Show details for a specific run
  export  Export run results to file
  init    Initialize a new reval project
  ui      Start the web interface
  version Show version information
Database Commands:
  db create  Create a new database
  db migrate Run database migrations
  db studio  Open Drizzle Studio
Quick Start:
1. reval init
# Initialize project
2. reval run
# Run your first benchmark
3. reval list
# View results
4. reval ui
# Explore in web interface
Run 'reval <command> --help' for detailed command usage
Documentation: https://github.com/your-org/reval"
`;
